lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -228,6 +228,11 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
     }
   }
 
+  late final ViewFocusBinding _viewFocusBinding = ViewFocusBinding(
+    viewManager: viewManager,
+    onViewFocusChange: invokeOnViewFocusChange,
```

Comment:
If this class exposed a stream of events, then you don't need to pass the `onViewFocusChange` through its constructor, and the `viewManager` can default to `EnginePlatformDispatcher.instance.viewManager`, so `ViewFocusBinding` can become a singleton again :P

Reply:
Hmmmm I find a bit weird the fact that EnginePlatformDispatcher depends on ViewFocusBinding and viceversa.

Reply:
? It already depends on it, I'm just saying that instead of:

```dart
final _viewFocusBinding = ViewFocusBinding(
  viewManager: viewManager,
  onViewFocusChange: invokeOnViewFocusChange,
)
```

you do:

```dart
final _viewFocusBinding = ViewFocusBinding(
  viewManager: viewManager,
); // or: ViewFocusBinding.instance;

// elsewhere

_viewFocusBinding.onViewFocusChange.listen((ViewFocusChange event) {
  // Do whatever we need with the event now...
});
```

That way you don't need to pass a tear-off into the ViewFocusBinding instance?

Reply:
> ? It already depends on it, I'm just saying that instead of:

How so? The view focus binding class  doesn't have any dependencies on `EnginePlatformDispatcher` only on `FlutterViewFocusManager`

Reply:
> The view focus binding class doesn't have any dependencies on `EnginePlatformDispatcher` only on `FlutterViewFocusManager`

Could you create the view focus binding class outside of the `EnginePlatformDispatcher`, without using the `EnginePlatformDispatcher` (to retrieve the `viewManager`? I think that's a fairly strong dependency :P